### PR TITLE
ToCode Solution: תרגול פקדים

### PR DESCRIPTION
**Pull request by wmashal**

### CODE REVIEW

File: 15-components-lab_v16/src/00-synctext.jsx [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+
+class MultiInput extends React.Component {
+  
+  constructor(props) {
+    super(props);
+    this.state = { textValue : "type sodsfmething..." };
+  }
+
+ write = (e) => {
+   this.setState({ textValue: e.target.value });
```
Comment (ynonp) [inline 15-components-lab_v16/src/00-synctext.jsx:12]: Did you try `this.setState(() => ({...}))` (Passing a function) ? Did it work? Can you explain why?

File: 15-components-lab_v16/src/00-synctext.jsx [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
- * all other input fields should show that updated value
- */
-
-import React from 'react';
-
-class MultiInput extends React.Component {
-}
-
+import React from 'react';
+
```
Comment (ynonp) [inline 15-components-lab_v16/src/00-synctext.jsx:2]: Works as expected. Good work

**Review by ynonp — COMMENTED** [review]
Works as expected. Good work

File: 15-components-lab_v16/src/01-timeconverter.jsx [hunk 3]
```diff
@@ -1,12 +1,40 @@
...
- * Component should show 3 input fields labeled hours, minutes and seconds.
- * Typing a value in either field updates the other two with corresponding values.
- */
-
-import React from 'react';
-
-export default React.createClass({
-});
+import React from 'react';
+
```
Comment (ynonp) [inline 15-components-lab_v16/src/01-timeconverter.jsx]: 1. Using `id` in React is considered bad practice. You never know how and when your component will be used. And for example if your component is used multiple times on the page, you get duplicate ids.

2. You saved 3 fields in `state`, but there's actually only one. The three numbers represent the same value in a different form (hours is just seconds divided by 3,600 for example). Please modify the code so you'll store only one item in `this.state`

File: 15-components-lab_v16/src/02-numberguessing.jsx [hunk 4]
```diff
@@ -1,17 +1,40 @@
...
+    
+   
+    constructor(props) {
+        super(props);
+        this.state = { randomNumber : Math.round(Math.random() * 1000) , message : "" };
+      }
+
+
+      check = (e) =>{
+        let guessedNo = parseInt(document.getElementById("guessedNo").value)
```
Comment (ynonp) [inline 15-components-lab_v16/src/02-numberguessing.jsx]: Don't use IDs nor getElementById in React. You never know how your components are going to be used.

Comment (ynonp) [reply 15-components-lab_v16/src/02-numberguessing.jsx]: Even querySelector is not recommended, as it's hard to limit its effect only to your rendered data.

In future lessons we'll learn how to query DOM elements from React. Until then use `value` and `onChange` to connect the input element to a value in state.

**Review by ynonp — CHANGES_REQUESTED** [review]
The solution works but not using best practices. Please review the comments and submit a fixed version.

**Comment by wmashal** [comment]
please run this command to make my color picker works
npm install react-color --save

**Pull request closed without merging**